asf_search/ASFProduct.py

```diff
@@ -41,7 +41,10 @@ def download(self, path: str, filename: str = None, session: ASFSession = None)
         :return: None
         """
         if filename is None:
-            filename = self.properties['fileName']
+            if self.properties['processingLevel'] == 'BURST':
```

Comment:
I think the burst extractor includes an actual filename in the content-disposition header, should probably check that value and use that instead, we definitely want to match whatever the extractor is using if possible